js/ui/appDisplay.js

```diff
@@ -1700,6 +1700,35 @@ const FolderView = new Lang.Class({
 
         this.addIcons();
         this.updateNoAppsLabelVisibility();
+
+        this.actor.connect('notify::mapped', Lang.bind(this, function(actor) {
+            // HACK!!! The only way to make the folder popover get the correct
+            // sizing is by removing and readding all the icons. To not hit the
+            // performance too badly, only do that when absolutely necessary.
+            if (!this.actor.mapped)
+                return;
+
+            this._redisplay();
+        }));
+
+        IconGridLayout.layout.connect('changed', Lang.bind(this, function() {
+            // AllView only checks for the toplevel icons. When dragging an icon
+            // from inside a folder to another folder, AllView won't be modified
+            // and this.iconsNeedRedraw() returns false. Thus, we have to manually
+            // update the folders too.
```

Comment:
Is this comment really true? From https://github.com/endlessm/gnome-shell/pull/165/commits/ac42f31aa306cf36ed1cc0a5d94fd1bb129a5f88#diff-7dfb7548c6314bf99dd14ab6cb89bc5aR282, it seems that `iconNeedRedraw()` would return `true` in that case.

Reply:
Forget this comment, I didn't notice this was inside FolderView when I commented